crates/telio-wg/src/wg.rs

```diff
@@ -717,8 +729,83 @@ impl State {
 
         self.interface = to.clone();
 
+        if !push {
+            // If we are pulling data from WG - update additional peer info
+            self.additional_peer_info = new_additional_peer_info;
+        }
+
         success
     }
+
+    fn compute_new_additional_peer_info(
+        &self,
+        to: &uapi::Interface,
+    ) -> HashMap<PublicKey, AdditionalPeerInfo> {
```

Comment:
`AdditionalPeerInfo` is correct that it's additional info, but it only takes part of the state, for example I would expect to find `last_endpoint_change` in there as well in that case.

My personal take would be to add `last_link_state` or something similar nearby as it was and not wrap it into this additional structure :). Or wrap it fully.